src/lists.c

```diff
@@ -1797,6 +1797,63 @@ void AsssListLevelCheck (
 }
 
 
+/****************************************************************************
+**
+*F  FuncSWAP_MAT_ROWS( <self>, <mat>, <row1>, <row2> )
+*/
+static Obj SwapMatRows;
+
+static Obj FuncSWAP_MAT_ROWS(Obj self, Obj mat, Obj row1, Obj row2)
+{
+    if (IS_POS_INTOBJ(row1) && IS_POS_INTOBJ(row2) && IS_PLIST(mat)) {
+        Int r1 = INT_INTOBJ(row1);
+        Int r2 = INT_INTOBJ(row2);
+        if (r1 <= LEN_PLIST(mat) && r2 <= LEN_PLIST(mat)) {
+            Obj row1list = ELM_PLIST(mat, r1);
+            if (!row1list)
+                ErrorMayQuit("Matrix Element: <mat>[%d] must have an assigned value",
+                             (Int)r1, 0);
+            Obj row2list = ELM_PLIST(mat, r2);
+            if (!row1list)
+                ErrorMayQuit("Matrix Element: <mat>[%d] must have an assigned value",
+                             (Int)r2, 0);
+
+            SET_ELM_PLIST(mat, r1, row2list);
+            SET_ELM_PLIST(mat, r2, row1list);
+        }
+        return 0;
+    }
+
+    return DoOperation3Args(SwapMatRows, mat, row1, row2);
+}
+
+
+/****************************************************************************
+**
+*F  FuncSWAP_MAT_COLS( <self>, <mat>, <col1>, <col2> )
+*/
+static Obj SwapMatCols;
+
+static Obj FuncSWAP_MAT_COLS(Obj self, Obj mat, Obj col1, Obj col2)
+{
+    if (IS_POS_INTOBJ(col1) && IS_POS_INTOBJ(col2) && IS_PLIST(mat)) {
+        Int c1 = INT_INTOBJ(col1);
+        Int c2 = INT_INTOBJ(col2);
+        Int nrows = LEN_PLIST(mat);
+        for (int r = 1; r <= nrows; r++) {
+            Obj row = ELM_LIST(mat, r);
```

Comment:
We could skip unbound rows...
```suggestion
            Obj row = ELM0_LIST(mat, r);
            if (!row) continue;
```